WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetails.swift

```diff
@@ -202,8 +225,12 @@ struct CouponDetails: View {
                     viewModel.loadCouponReport()
                 })
             }
-            .sheet(isPresented: $viewModel.showingEditCoupon) {
-                AddEditCoupon(addEditCouponViewModel)
+            .sheet(isPresented: $viewModel.showingEditCoupon, onDismiss: {
+                // this is just a test to compare calling the AddEditCoupon view from a callback
+                // and from the UIViewControllerRepresentable bridge implementation below
+                onEditCoupon(addEditCouponViewModel)
+            }) {
+                AddEditCouponHostingControllerBridge(viewModel: addEditCouponViewModel)
             }
```

Comment:
Since we want to present the `AddEditCoupon` view the UIKit way, we should remove this and instead call the `onEditCoupon` closure when we need to show the view. Show instead of this:

https://github.com/woocommerce/woocommerce-ios/blob/91acf263d2ffbbc52c7e36abc4c0e307df86b04d/WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetails.swift#L83-L88

We should trigger the closure like so:

```Swift
buttons.append(contentsOf: [ 
     .default(Text(Localization.editCoupon), action: { 
         ServiceLocator.analytics.track(.couponDetails, withProperties: ["action": "tapped_edit"]) 
         onEditCoupon(addEditCouponViewModel)
     }) 
 ])
```

Reply:
Oh, makes perfect sense! I did that in the first place, but since I was comparing each of the three implementations, I decided to call everything at once so I could compare how each one was behaving! The callback is now linked within the button action! Commit: https://github.com/woocommerce/woocommerce-ios/pull/6889/commits/cc7a93d051768e3431e0810edf95ba1f34c49fc1